pkg/nodesconfig/nodesconfig.go

```diff
@@ -0,0 +1,194 @@
+package nodesconfig
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	configV1 "github.com/openshift/api/config/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Builder provides a struct for nodesConfig object from the cluster and a nodesConfig definition.
+type Builder struct {
+	// nodesConfig definition, used to create the nodesConfig object.
+	Definition *configV1.Node
+	// Created nodesConfig object.
+	Object *configV1.Node
+	// api client to interact with the cluster.
+	apiClient goclient.Client
+	// Used in functions that define or mutate clusterOperator definition. errorMsg is processed before the
+	// ClusterOperator object is created.
+	errorMsg string
+}
+
+// Pull retrieves an existing nodesConfig object from the cluster.
+func Pull(apiClient *clients.Settings, nodesConfigObjName string) (*Builder, error) {
+	if apiClient == nil {
+		glog.V(100).Infof("The apiClient is empty")
+
+		return nil, fmt.Errorf("nodesConfig Config 'apiClient' cannot be empty")
+	}
+
+	if nodesConfigObjName == "" {
+		glog.V(100).Infof("The name of the nodesConfig is empty")
+
+		return nil, fmt.Errorf("nodesConfig 'nodesConfigObjName' cannot be empty")
+	}
+
+	glog.V(100).Infof(
```

Comment:
Please move this logs to the first like of this function

Reply:
not sure what you ask about, but pls check